types/underscore/index.d.ts

```diff
@@ -869,8 +875,18 @@ declare module _ {
          * @param lists The lists to unzip.
          * @returns The unzipped version of `lists`.
          **/
+        unzip<T, U, V>(lists: List<[T, U, V]>): [T[], U[], V[]];
+        unzip<T, U>(lists: List<[T, U]>): [T[], U[]];
+        unzip<T>(lists: List<[T]>): [T[]]; // eslint-disable-line no-single-element-tuple-type
+        unzip<T>(lists: List<List<T>>): T[][];
```

Comment:
Don't these two lines mean roughly the same? Why would you need both?

Reply:
The first line means "a list of length 1 that contains a list of T", whereas the second line means "a list of lists of T". The first type would catch an error where a user tries to do a destructuring assignment into too many variables, for example:

```
const [a, b] = _.unzip([[1], [2]]); // Tuple type '[number]' of length '1' has no element at index '1'. (2493)
const [a, b] = _.unzip([[1, true], [2, false]]); // all is well
```